libstratum/EthStratumClient.cpp

```diff
@@ -28,7 +28,8 @@ static void diffToTarget(uint32_t *target, double diff)
 
 
 EthStratumClient::EthStratumClient(Farm* f, MinerType m, string const & host, string const & port, string const & user, string const & pass, int const & retries, int const & worktimeout, int const & protocol, string const & email)
-	: m_socket(m_io_service)
+        :       m_socket(m_io_service),
+	        m_worktimer(m_io_service, boost::posix_time::seconds(99999)) // effectively set it to infinity
```

Comment:
99999 not needed. The constructor without any time will not start the timer. See: http://www.boost.org/doc/libs/1_66_0/doc/html/boost_asio/reference/basic_deadline_timer/basic_deadline_timer/overload1.html.

Reply:
Ah! Ok, I'll change that